koin-projects/koin-core/src/test/kotlin/org/koin/dsl/BeanLifecycleTest.kt

```diff
@@ -109,4 +111,82 @@ class BeanLifecycleTest {
         scopeInstance.close()
         assertEquals("release", result)
     }
+
+    /**
+     * https://github.com/InsertKoinIO/koin/issues/492
+     */
+    @Test
+    fun `closing scope A1 does not affect bean definitions of scope A2 when both scope instances have the same scope name A`() {
+        val koin = koinApplication {
+            modules(module {
+                scope(named("A")) {
+                    scoped { Simple.Component1() }
+                }
+            })
+        }.koin
+
+        val scopeA1 = koin.createScope("123", named("A"))
+        val scopeA2 = koin.createScope("234", named("A"))
+
+        val scopeA1Value = scopeA1.get<Simple.Component1>()
+        val scopeA2Value = scopeA2.get<Simple.Component1>()
+        assertNotNull(scopeA1Value)
+        assertNotNull(scopeA2Value)
+        assertNotEquals(scopeA1Value, scopeA2Value)
+
+        scopeA1.close()
+
+        assertEquals(scopeA2Value, scopeA2.get<Simple.Component1>())
+        try {
+            scopeA1.get<Simple.Component1>()
+            fail("Should not be able to resolve the instance from scope A because scope A is closed.")
+        } catch(e: NoBeanDefFoundException) {
```

Comment:
I think you could replace try/catch block, only using `@Test(expected = NoBeanDefFoundException::class)`.

Reply:
This might lead to a false positive because `Simple.Component1` is resolved twice and should only throw the exception at the second call. So the exception is not expected for the first retrieval of `Simple.Component1`.